src/js/hds-cc/hds-cc.js

```diff
@@ -48,66 +180,118 @@ class HdsCc {
-      // If cookie parsing fails, show banner
-      console.log(`Cookie parsing unsuccessful:\n${err}`);
+      console.error(`Cookie parsing unsuccessful:\n${err}`);
       return false;
     }
+  }
+
+
 
-    return cookie;
+  /**
+   * Removes the banner and related elements.
+   * @private
+   * @memberof ClassName
+   * @function #removeBanner
+   */
+  #removeBanner() {
+    // Remove banner size observer
+    if (this.resizeReference.resizeObserver && this.resizeReference.bannerHeightElement) {
+      this.resizeReference.resizeObserver.unobserve(this.resizeReference.bannerHeightElement);
+    }
+    // Remove banner elements
+    if (this.bannerElements.bannerContainer) {
+      this.bannerElements.bannerContainer.remove();
+    }
+    if (this.bannerElements.spacer) {
+      this.bannerElements.spacer.remove();
+    }
+    // Remove scroll-margin-bottom variable from all elements inside the contentSelector
+    document.documentElement.style.removeProperty('--hds-cookie-consent-height');
   }
 
-  async setCookie(cookieData) {
-    console.log('setCookie', cookieData);
+
+  /**
+   * Sets a cookie with the provided data.
+   * @private
+   * @param {Object} cookieData - The data to be stored in the cookie.
+   */
+  #setCookie(cookieData) {
+    // console.log('#setCookie', cookieData);
     const expiryDate = new Date();
-    expiryDate.setDate(expiryDate.getDate() + this.COOKIE_DAYS);
-    document.cookie = serialize(this.COOKIE_NAME, JSON.stringify(cookieData), {expires: expiryDate});
+    expiryDate.setDate(expiryDate.getDate() + this.#COOKIE_DAYS);
+    document.cookie = serialize(this.cookie_name, JSON.stringify(cookieData), { expires: expiryDate });
   }
 
-  saveAcceptedGroups(cookieSettings, acceptedGroupNames = [], showBanner = false) {
-    console.log('Accepting cookies from groups:', acceptedGroupNames);
+
+  /**
+   * Saves the accepted cookie groups to cookie, unsets others.
+   * @private
+   * @param {Object} cookieSettings - Site specific settings
+   * @param {Array} acceptedGroupNames - The names of the accepted cookie groups.
+   * @param {boolean} showBanner - Whether to show the banner or not.
+   */
+  #saveAcceptedGroups(cookieSettings, acceptedGroupNames = [], showBanner = false) {
+    // console.log('Saving accepted cookie groups:', acceptedGroupNames);
 
     const acceptedGroups = {};
-    // Required should not come in every case; the category hash might have changed
-    cookieSettings.requiredCookies.groups.forEach(group => {
-      if (acceptedGroupNames.includes(group.groupId)) {
-        acceptedGroups[group.groupId] = group.checksum;
-      }
-    });
 
-    // Add accepted group names to acceptedGroups, assume others to be declined (overrides previous selections)
-    cookieSettings.optionalCookies.groups.forEach(group => {
+    // Find group checksums for accepted groups
+    const bothGroups = [...cookieSettings.requiredCookies.groups, ...cookieSettings.optionalCookies.groups];
```

Comment:
Could be more accurate as 'allGroups' or 'bothCategories'

Reply:
Changed to allGroups